scipy/sparse/_lil.py

```diff
@@ -286,7 +286,7 @@ def _mul_scalar(self, other):
             res_dtype = upcast_scalar(self.dtype, other)
 
             new = self.copy()
-            new = new.astype(res_dtype)
+            new = new.astype(res_dtype, copy=False)
```

Comment:
Maybe we could do something fancy here by avoiding the explicit copy when the dtype conversion would copy anyway.